src/common/src/util/value_encoding/column_aware_row_encoding.rs

```diff
@@ -105,8 +106,77 @@ impl RowEncoding {
     }
 }
 
+trait ValueRowSerializer: Clone {
+    fn new(column_ids: &[ColumnId]) -> impl ValueRowSerializer;
+    fn serialize(&self, row: impl Row) -> Vec<u8>;
+}
+
+trait ValueRowDeserializer: Clone {
+    fn new(column_ids: &[ColumnId], schema: Arc<[DataType]>) -> impl ValueRowDeserializer;
+    fn deserialize(&self, encoded_bytes: &[u8]) -> Result<Vec<Datum>>;
+}
+
+pub trait ValueRowSerde: Clone {
+    fn new(column_ids: &[ColumnId], schema: Arc<[DataType]>) -> Self;
+    fn serialize(&self, row: impl Row) -> Vec<u8>;
+    fn deserialize(&self, encoded_bytes: &[u8]) -> Result<Vec<Datum>>;
+}
```

Comment:
Why not `ValueRowSerdeCreate` `ValueRowSerializer` `ValueRowDeserializer`? If we really need ONE trait as public interface, we can do `ValueRowSerde: ValueRowSerdeCreate + ValueRowSerializer + ValueRowDeserializer` and simply forward the method calls?

Another question is why `ValueRowSerializer::new` and `ValueRowDeserializer::new` return `impl Xxx` but `ValueRowSerde::new` returns `Self`. IIUC returning `Self` is OK for all of these traits?

Reply:
+1. Actually, we've done this refactor before (while removed later), and a pattern of trait definition was suggested here -> https://github.com/risingwavelabs/risingwave/pull/3925#issue-1305875710. You may take it as a reference. 🥰